docs/runbooks/cloud9/cloud9_init.sh

```diff
@@ -1,45 +1,95 @@
 #! /bin/bash
+function main() {
+  install_tools
+  infer_account ${1:?}
+  add_rds_sgs ${1:?}
+  postgresql ${1:?}
+  setup_info ${1:?}
+}
 
-for arg in "$@"
-do
-    if [ "$arg" == "--help" ] || [ "$arg" == "-h" ]
-    then
-        echo "Set up cloud9 environment with some tools. takes an environment name as an argument"
-        exit 0
-    fi
-done
-
-# install php dependencies
-sudo yum remove php* -y
-sudo yum install php73 php73-pdo php73-pgsql postgresql jq -y
-
-# install terraform
-export TF_VERSION=0.12.10
-export TF_SHA256SUM=2215208822f1a183fb57e24289de417c9b3157affbe8a5e520b768edbcb420b4
-export TF_VAR_default_role=operator
-export TF_VAR_management_role=operator
-export TF_CLI_ARGS_init="-upgrade=true  -reconfigure"
-export TF_WORKSPACE=$1
-curl -sfSO https://releases.hashicorp.com/terraform/${TF_VERSION}/terraform_${TF_VERSION}_linux_amd64.zip
-echo "${TF_SHA256SUM} terraform_${TF_VERSION}_linux_amd64.zip" > SHA256SUMS
-sha256sum -c --status SHA256SUMS
-sudo unzip -o terraform_${TF_VERSION}_linux_amd64.zip -d /bin
-terraform -version
-
-# set db env vars
-export ENV_NAME=$(echo "$TF_WORKSPACE" | awk '{print tolower($0)}')
-export AWS_DEFAULT_REGION=eu-west-1
-export OPG_REFUNDS_DB_APPLICATIONS_HOSTNAME=$(aws rds describe-db-clusters --db-cluster-identifier applications-$ENV_NAME | jq -r .'DBClusters'[0].'Endpoint')
-export OPG_REFUNDS_DB_APPLICATIONS_PORT=5432
-export CASES_DB_ENDPOINT=$(aws rds describe-db-clusters --db-cluster-identifier caseworker-$ENV_NAME | jq -r .'DBClusters'[0].'Endpoint')
-export OPG_REFUNDS_DB_FINANCE_HOSTNAME=$CASES_DB_ENDPOINT
-export OPG_REFUNDS_DB_FINANCE_PORT=5432
-export OPG_REFUNDS_DB_SIRIUS_HOSTNAME=$CASES_DB_ENDPOINT
-export OPG_REFUNDS_DB_SIRIUS_PORT=5432
-export OPG_REFUNDS_DB_MERIS_HOSTNAME=$CASES_DB_ENDPOINT
-export OPG_REFUNDS_DB_MERIS_PORT=5432
-export POSTGRES_USER=root
-export PGPASSWORD=$(aws secretsmanager get-secret-value --secret-id $TF_WORKSPACE/postgres_password | jq -r .'SecretString')
-
-# Set cloud9 IP for ingress
-export CLOUD9_IP=$(ip route get 1 | awk '{print $NF;exit}')
+
+function install_tools() {
+  sudo yum install postgresql jq -y
+}
+
+function infer_account() {
+  case "${1:?}" in
+  'preproduction')
+  export ACCOUNT="${1:?}"
+  ;;
+  'production')
+  export ACCOUNT="${1:?}"
+  ;;
+  *)
+  export ACCOUNT="development"
+  ;;
+  esac
+}
+
+function add_rds_sgs() {
+  local current_sg_name
+  local current_sg_id
+  local environment_name=$(echo ${1:?} | awk '{print tolower($0)}')
+  local applications_rds_client_sg
+  local caseworker_rds_client_sg
+  local CURRENT_SG_IDS=()
+
+  echo "getting current sg names and instance id..."
+  current_sg_names=$(curl -s http://169.254.169.254/latest/meta-data/security-groups)
```

Comment:
Are these IP addresses configurable or are they a known static address?

Reply:
It's static, this is the aws metadata service https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html